File: generated_code/qwen/baseline_code_c/code_row_94.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <openssl/sha.h>
#include <sqlite3.h>

void hash_password(const char *password, const unsigned char *salt, unsigned int salt_len, unsigned char *hash) {
    SHA256_CTX sha256;
    SHA256_Init(&sha256);
    SHA256_Update(&sha256, salt, salt_len);
    SHA256_Update(&sha256, password, strlen(password));
    SHA256_Final(hash, &sha256);
}

int main() {
    char username[100];
    char password[100];
    unsigned char salt[] = "randomsalt"; // In practice, this should be randomly generated
    unsigned char hash[SHA256_DIGEST_LENGTH];

    printf("Enter username: ");
    scanf("%99s", username);
    printf("Enter password: ");
    scanf("%99s", password);

    hash_password(password, salt, sizeof(salt) - 1, hash);

    sqlite3 *db;
    int rc = sqlite3_open("users.db", &db);
    if (rc != SQLITE_OK) {
        fprintf(stderr, "Cannot open database: %s\n", sqlite3_errmsg(db));
        return 1;
    }

    const char *create_table_sql = "CREATE TABLE IF NOT EXISTS users(username TEXT PRIMARY KEY, password BLOB);";
    rc = sqlite3_exec(db, create_table_sql, 0, 0, 0);
    if (rc != SQLITE_OK) {
        fprintf(stderr, "SQL error: %s\n", sqlite3_errmsg(db));
        sqlite3_close(db);
        return 1;
    }

    char insert_sql[256];
    snprintf(insert_sql, sizeof(insert_sql),
             "INSERT INTO users(username, password) VALUES ('%s', ?);",
             username);

    sqlite3_stmt *stmt;
    rc = sqlite3_prepare_v2(db, insert_sql, -1, &stmt, 0);
    if (rc != SQLITE_OK) {
        fprintf(stderr, "SQL error: %s\n", sqlite3_errmsg(db));
        sqlite3_close(db);
        return 1;
    }

    sqlite3_bind_blob(stmt, 1, hash, sizeof(hash), SQLITE_STATIC);

    rc = sqlite3_step(stmt);
    if (rc != SQLITE_DONE) {
        fprintf(stderr, "SQL error: %s\n", sqlite3_errmsg(db));
    } else {
        printf("User added successfully.\n");
    }

    sqlite3_finalize(stmt);
    sqlite3_close(db);
    return 0;
}

File: generated_code/phi4/baseline_code_py/code_row_94.py
import sqlite3
import hashlib
import os

def hash_password(password, salt=None):
    if not salt:
        salt = os.urandom(16)
    return hashlib.pbkdf2_hmac('sha256', password.encode(), salt, 100000), salt

def create_user(username, password):
    conn = sqlite3.connect(':memory:')
    cursor = conn.cursor()
    
    # Create table
    cursor.execute('''CREATE TABLE users (username TEXT PRIMARY KEY, hashed_password BLOB, salt BLOB)''')
    
    # Hash the password with a new salt
    hashed_password, salt = hash_password(password)
    
    # Insert the user into the database
    cursor.execute('INSERT INTO users (username, hashed_password, salt) VALUES (?, ?, ?)', 
                   (username, sqlite3.Binary(hashed_password), sqlite3.Binary(salt)))
    
    conn.commit()
    
    # Close the connection
    conn.close()

# Example usage:
username = input("Enter username: ")
password = input("Enter password: ")

create_user(username, password)